internal/backend/sftp/sftp.go

```diff
@@ -594,3 +596,64 @@ func (r *SFTP) Warmup(_ context.Context, _ []backend.Handle) ([]backend.Handle,
 	return []backend.Handle{}, nil
 }
 func (r *SFTP) WarmupWait(_ context.Context, _ []backend.Handle) error { return nil }
+
+// Mkdir creates a directory at the specified path. It returns an error if the directory
+// already exists, if the parent directory doesn't exist, or if path is an existing file.
+func (r *SFTP) Mkdir(path string) error {
+	if err := r.clientError(); err != nil {
+		return err
+	}
+
+	// Check if path already exists
+	fi, err := r.c.Stat(path)
+	if err == nil {
+		if fi.IsDir() {
+			return &os.PathError{Op: "mkdir", Path: path, Err: os.ErrExist}
+		}
+		return &os.PathError{Op: "mkdir", Path: path, Err: fmt.Errorf("path exists but is a file")}
+	}
+	if !r.IsNotExist(err) {
+		return errors.Wrap(err, "Stat")
+	}
+
+	// Create the directory
+	err = r.c.Mkdir(path)
+	if err != nil {
+		return errors.Wrap(err, "Mkdir")
+	}
+
+	// Set directory permissions
+	return r.c.Chmod(path, r.Modes.Dir)
+}
+
+// MkdirAll creates a directory named path, along with any necessary parents.
+// If path is already a directory, MkdirAll does nothing and returns nil.
+// If path exists but is a file, an error is returned.
+func (r *SFTP) MkdirAll(path string) error {
```

Comment:
Why does this method behave rather different from the `MkdirAll` implementation in the sftp package, when the goal of this PR is to just fix the directory permissions?

The sftp package checks if the parent folder exists, if no then it recursively calls itself, and afterwards creates the requested directory. That requires much fewer calls to the server than this implementation.